src/textual/app.py

```diff
@@ -562,7 +562,20 @@ def __init__(
         """Number of lines to scroll in the Y direction with wheel or trackpad."""
 
         self._installed_screens: dict[str, Screen | Callable[[], Screen]] = {}
+        for var_name, dct in (("SCREENS", self.SCREENS), ("MODES", self.MODES)):
+            for screen_name, v in dct.items():
+                if (
```

Comment:
I think we can simplify a little. The only legal types are a `str` or anything callable.

Maybe this?

```python
if not (isinstance(screen_name, str) or callable(screen_name)):
```